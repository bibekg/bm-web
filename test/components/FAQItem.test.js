import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import FAQItem from 'components/FAQItem'

describe('<FAQItem />', () => {
  const spy = jest.spyOn(FAQItem.prototype, 'toggleAccordion')
  const wrapper = shallow(<FAQItem headerText="header" innerText="inner" />)

  test('should render correctly closed', () => {
    expect(wrapper).toMatchSnapshot()

    // check that header text is present and correct
    expect(
      wrapper
        .childAt(0) // AccordionHeaderDiv
        .childAt(0) // Subtitle
        .children()
        .text()
    ).toEqual('header')

    // check that switch is rotated correctly
    expect(wrapper.childAt(0).childAt(1)).toHaveStyleRule('transform', 'rotate(0deg)')

    // check that DropdownDiv isn't currently rendered (i.e. AnimateHeight's height is 0)
    expect(wrapper.childAt(1).prop('height')).toEqual(0)
  })

  test('should unfold on click and render correctly', () => {
    wrapper.childAt(0).simulate('click')
    expect(spy).toHaveBeenCalled()
    wrapper.update()
    expect(wrapper).toMatchSnapshot()

    // check that switch is rotated correctly
    expect(wrapper.childAt(0).childAt(1)).toHaveStyleRule('transform', 'rotate(180deg)')

    // check that DropdownDiv is now rendered
    expect(wrapper.childAt(1).prop('height')).toEqual('auto')

    // check that inner text is present and correct
    expect(
      wrapper
        .childAt(1) // AnimateHeight
        .childAt(0) // DropdownDiv
        .childAt(0) // Text
        .children()
        .text()
    ).toEqual('inner')
  })

  test('should refold and render correctly on click', () => {
    wrapper.childAt(0).simulate('click')
    expect(spy).toHaveBeenCalledTimes(2)
    wrapper.update()
    expect(wrapper).toMatchSnapshot()

    // check that DropdownDiv is now unrendered
    expect(wrapper.childAt(1).prop('height')).toEqual(0)
  })
})
