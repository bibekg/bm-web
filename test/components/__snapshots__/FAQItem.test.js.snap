// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQItem /> should refold and render correctly on click 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <Styled(styled.h1)
      align="left"
    >
      header
    </Styled(styled.h1)>
    <styled.div
      isOpen={false}
    >
      <ExpandMoreIcon
        fill="#76AFFB"
        size={40}
      />
    </styled.div>
  </styled.div>
  <AnimateHeight
    animateOpacity={false}
    animationStateClasses={
      Object {
        "animating": "rah-animating",
        "animatingDown": "rah-animating--down",
        "animatingToHeightAuto": "rah-animating--to-height-auto",
        "animatingToHeightSpecific": "rah-animating--to-height-specific",
        "animatingToHeightZero": "rah-animating--to-height-zero",
        "animatingUp": "rah-animating--up",
        "static": "rah-static",
        "staticHeightAuto": "rah-static--height-auto",
        "staticHeightSpecific": "rah-static--height-specific",
        "staticHeightZero": "rah-static--height-zero",
      }
    }
    applyInlineTransitions={true}
    delay={0}
    duration={250}
    easing="ease"
    height={0}
    style={Object {}}
  >
    <styled.div>
      <styled.p
        paragraph={true}
      >
        inner
      </styled.p>
    </styled.div>
  </AnimateHeight>
</styled.div>
`;

exports[`<FAQItem /> should render correctly closed 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <Styled(styled.h1)
      align="left"
    >
      header
    </Styled(styled.h1)>
    <styled.div
      isOpen={false}
    >
      <ExpandMoreIcon
        fill="#76AFFB"
        size={40}
      />
    </styled.div>
  </styled.div>
  <AnimateHeight
    animateOpacity={false}
    animationStateClasses={
      Object {
        "animating": "rah-animating",
        "animatingDown": "rah-animating--down",
        "animatingToHeightAuto": "rah-animating--to-height-auto",
        "animatingToHeightSpecific": "rah-animating--to-height-specific",
        "animatingToHeightZero": "rah-animating--to-height-zero",
        "animatingUp": "rah-animating--up",
        "static": "rah-static",
        "staticHeightAuto": "rah-static--height-auto",
        "staticHeightSpecific": "rah-static--height-specific",
        "staticHeightZero": "rah-static--height-zero",
      }
    }
    applyInlineTransitions={true}
    delay={0}
    duration={250}
    easing="ease"
    height={0}
    style={Object {}}
  >
    <styled.div>
      <styled.p
        paragraph={true}
      >
        inner
      </styled.p>
    </styled.div>
  </AnimateHeight>
</styled.div>
`;

exports[`<FAQItem /> should unfold on click and render correctly 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <Styled(styled.h1)
      align="left"
    >
      header
    </Styled(styled.h1)>
    <styled.div
      isOpen={true}
    >
      <ExpandMoreIcon
        fill="#76AFFB"
        size={40}
      />
    </styled.div>
  </styled.div>
  <AnimateHeight
    animateOpacity={false}
    animationStateClasses={
      Object {
        "animating": "rah-animating",
        "animatingDown": "rah-animating--down",
        "animatingToHeightAuto": "rah-animating--to-height-auto",
        "animatingToHeightSpecific": "rah-animating--to-height-specific",
        "animatingToHeightZero": "rah-animating--to-height-zero",
        "animatingUp": "rah-animating--up",
        "static": "rah-static",
        "staticHeightAuto": "rah-static--height-auto",
        "staticHeightSpecific": "rah-static--height-specific",
        "staticHeightZero": "rah-static--height-zero",
      }
    }
    applyInlineTransitions={true}
    delay={0}
    duration={250}
    easing="ease"
    height="auto"
    style={Object {}}
  >
    <styled.div>
      <styled.p
        paragraph={true}
      >
        inner
      </styled.p>
    </styled.div>
  </AnimateHeight>
</styled.div>
`;
